### ISSUE TITLE
CachingProcessingComponentManager is breaking application's cache manager. [CARROT-919]

### ISSUE THREAD

**Issue by dweiss** [carrot2/carrot2#1117]
When using CachingProcessingComponentManager, it tries to share the CacheManager, instead of creating a new CacheManager. This is really dangerous.

If the application creates a CacheManager, before instantiating CachingProcessingComponentManager, then Carrot is using the application's cache manager and vice-verse.
 

```Java
public CachingProcessingComponentManager(IProcessingComponentManager delegate,
        Class<? extends IProcessingComponent>... cachedComponentClasses)
    {
        this.delegate = delegate;
        this.cachedComponentClasses = ImmutableSet.copyOf(cachedComponentClasses);

        // Initialize cache
        try
        {
            cacheManager = CacheManager.create(new ClassResource(
                CachingProcessingComponentManager.class, "/controller-ehcache.xml")
                .open());
        }
        catch (IOException e)
        {
            throw new ComponentInitializationException("Could not initalize cache.", e);
        }

        // Create a new cache for each controller instance.
        final String cacheName = "CachingProcessingComponentManagerCache"
            + new Random().nextInt();
        cacheManager.addCache(cacheName);
        dataCache = new SelfPopulatingCache(cacheManager.getCache(cacheName),
            new CachedDataFactory());
    }
```

Refer: Spring's EhCacheManagerFactoryBean to instantiate a new CacheManager (not shared).

---
Issue: CARROT-919 (migrated from JIRA), created by Daniel Muszynski, resolved May 31 2012

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

I'll look into this. I don't think we will release 3.5.4 though – I'm working on 3.6.0 and this removes ehcache entirely so the problem is gone.

**Comment by dweiss** [comment]
Comment by Daniel Muszynski (migrated from JIRA)

Should have been:

new CacheManager(new ClassResource(CachingProcessingComponentManager.class, "/controller-ehcache.xml").open()) 

instead of:

CacheManager.create(new ClassResource(CachingProcessingComponentManager.class, "/controller-ehcache.xml").open())

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

Confirmed, CacheManager.create\* acquires a singleton and may clash with other stuff (from the same classloader scope).

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

Fixed on bugfix/3.5.x branch. Problem does not exist on master (google cache is used instead).
